code/game/objects/structures/dropship_equipment.dm

```diff
@@ -1039,3 +1042,328 @@
+		return
+	if(origin.hooked)
+		playsound(origin, 'sound/machines/buzz-sigh.ogg', 25)
+		return
+
+	C.playsound_local(origin, 'sound/effects/binoctarget.ogg', 35)
+
+	if(origin.unlocked)
+		origin.pre_retract()
+		return
+
+
+	if(!do_after(C, 5 SECONDS, TRUE, remote_eye, BUSY_ICON_BAR))
+		return
+
+	origin.deploy_rope(target_turf)
+	playsound(origin, 'sound/machines/chime.ogg', 50, FALSE)
+
+
+
+//The actual system you put on the tadpole
+/obj/structure/dropship_equipment/rappel_system
+	equip_category = DROPSHIP_CREW_WEAPON
+	name = "rappel deployment system"
+	desc = "A system that deploys rappel ropes to go up or down fast, without the need for the Tadpole to land. You need to designate the rappel point at the navigation computer."
+	dropship_equipment_flags = IS_INTERACTABLE
+	icon_state = "rappel_hatch_locked"
+	point_cost = 300
+	coverage = 0
+	max_integrity = 300
+	anchored = TRUE
+	density = FALSE
+	///The rappeling rope we use
+	var/obj/effect/rappel_rope/tadpole/rope
+	///Whether marines can use the rope or not (i.e, if the rope is deployed)
+	var/unlocked = FALSE
+	///Whether xenos have temporarily disabled the system or not
+	var/disabled = FALSE
+	///Whether the system's cord needs replacing or not
+	var/cord_damaged = FALSE
+	///Smoke particle holder for when it's disabled
+	var/obj/effect/abstract/particle_holder/disabled_smoke
+	///Whether the system is currently retracting a rope or not
+	var/retracting = FALSE
+	///Whether a xeno is currently disabling the system or not
+	var/hooked = FALSE
+
+/obj/structure/dropship_equipment/rappel_system/Initialize()
+	. = ..()
+	rope = new
+	rope.forceMove(src)
+	rope.parent_system = src
+
+
+/obj/structure/dropship_equipment/rappel_system/Destroy()
+	. = ..()
+	QDEL_NULL(rope)
+
+
+/obj/structure/dropship_equipment/rappel_system/examine(mob/user)
+	. = ..()
+	if(disabled)
+		. += "It's smoking. It'll take some time for its self-repair function to kick in."
+	if(cord_damaged)
+		. += "The rappel's cord is completely broken and needs replacing."
+
+
+///Human interaction with the rappel system; this is how people rappel down
+/obj/structure/dropship_equipment/rappel_system/attack_hand(mob/living/carbon/human/user)
+	if(disabled)
+		balloon_alert(user, "The system is disabled!")
+		return
+
+	if(cord_damaged)
+		balloon_alert(user, "The cord needs replacing!")
+		return
+
+	if(!is_reserved_level(z))
+		balloon_alert(user, "You are not in-flight!")
+		return
+
+	if(!rope || !unlocked)
+		balloon_alert(user, "No rappel deployed!")
+		return
+
+	if(retracting)
+		balloon_alert(user, "The rappel is currently retracting!")
+		return
+
+	var/turf/target_turf = get_turf(rope)
+	if(target_turf.density)
+		balloon_alert(user, "You can't rappel into a wall!")
+		return
+
+	var/area/target_area = get_area(target_turf)
+	if(target_area.ceiling >= CEILING_DEEP_UNDERGROUND)
+		balloon_alert(user, "The rappel is too deep underground!")
+		return
+
+	rope.icon_state = "rope_rappeling"
```

Comment:
No icon state change outside of update_icon_state()

Also tgmc is oop design, stop accessing directly in rope vars where that should be function call